src/ucp/rndv/proto_rndv.c

```diff
@@ -1071,3 +1071,13 @@ void ucp_proto_rndv_bulk_request_init_lane_idx(
 
     req->send.multi_lane_idx = lane_idx - 1;
 }
+
+void ucp_proto_rndv_ppln_abort(ucp_request_t *req, ucs_status_t status)
+{
+    /* FIXME: Proper abort functionality is not implemented yet.
+     * This stub function is used to advertise error-handling capability for
+     * pipeline protocols, but proper implementation is to be done */
+    ucs_warn("aborting ppln request %p with status %s", req,
```

Comment:
WDYT about making warning more informative and describe what it can lead to? 
```suggestion
    ucs_warn("rndv ppln request %p was aborted with status %s. "
                    "This may lead to memory leaks since rndv ppln abortion "
                    "mocking was enabled", req,
```

Reply:
My understanding is that it may lead to memory corruption rather than memory leak.
Because when we detect that EP is in failed state, we should invalidate the memh we exposed to prevent stale writes to it. Since we don't do invalidation yet, these writes may appear and damage chunk elements already used for different purpose.
So in worst case the same element is reused for some other operation and we can get data corrupted.
I can update the warning message with these details if you agree

Reply:
Yeah, AFAIK that is true too. I thought about memory leaks because we are using mempools and I am not sure that we are returning them back in case of this abortion stub calling. 

Mentioning memory corruption is OK from my point of view.

Reply:
pls use `ucs_diag` instead of `ucs_warn`

Reply:
> pls use `ucs_diag` instead of `ucs_warn`

why? AFAIU this callback is not expected to be called, IMO warning is ok here.

Reply:
it can be called during error handling flow 
(even without invalidation better not to print something to warning)

Reply:
done

Reply:
It can be, but unexpected and may lead to undefined behaviour.
Is invalidation the only missed thing on this flow? if so then just need to move this warning under "connected-to-iface" condition.

Reply:
i had am impression that invalidation is the only missing piece (ilia pls confirm).
I'd still leave it as `diag`, because even with `connect-to-iface` invalidation may not be really required. 
The point is that this how things worked in protov1 and users were satisfied with this behavior. Also we already protected  this (not fully complete) code with env var to be able to disable such behavior.

Reply:
Yes, I confirm that only invalidation is missing
IMO it's not really a big deal, as subsequent commit will remove this code anyway..

Reply:
+1 for leaving as warning (or at least debug message)

Reply:
> +1 for leaving as warning (or at least debug message)

we are not going to remove it. We are duscussing the log level to be used

Reply:
if you are sure that `connect-to-ep` works as expected, then this message will only confuse with any level.

I see 2 potential issues here:
1) completion if single fragment might be not enough to complete full ppl protocol. It may be relevant for any type of transport. (@iyastreb confirmed that this is not the case).
2) invalidation flow is valid only for `connect-to-iface` transport.
So, 
1) that's the reason to have it for `connect-to-ep`?
2) invalidation flow may lead to hardly reproducible data corruption bug, IMO warning is preferable because on re-run (with log level diag) error flow may be not invoked at all or invoked but without data corruption.

Reply:
> > +1 for leaving as warning (or at least debug message)
> 
> we are not going to remove it. We are duscussing the log level to be used

I understand, but I think that we need to print a warning message instead of diag to explicitly show user that something went wrong and call that was not supposed to be called was called. FMPOV if we didn't print warning in that case in protov1 it doesn't mean that we should keep this behavior now.

Reply:
let's discuss it on meeting, this discussion becomes too long